### ISSUE TITLE
Validating tuples

### ISSUE THREAD

**Issue by doomsower** [pleerock/class-validator#117]
I need to validate user input that has arrays of tuples:
```ts
type Coordinate = [number, number, number]; // [longitude, latitude, altitude]

class Region {
  ...
  shape: Coordinate[];
}
```

This question splits into two:
1. I created custom validator decorator `@IsCoordinate` which validates tuples. However, when I use it like this:
    ```ts
    @IsCoordinate({ each: true })
    shape: Coordinate[];
    ```

    I get en error on empty arrays of tuples, e.g. `const region = { shape: [], ...rest }`. It seems that validator treats outer array as inner and tries to validate array of tuples as if it is a single tuple.

2. How can I validate each of tuple's positions separately and get validation error related to this field? With `@isCoordinate` I can only validate them in one piece, and the only way to distinguish if it is latitude incorrect or longitude incorrect is to use message string

**Comment by 19majkel94** [comment]
> `{ each: true }`

Why you do this? In this case it doesn't validate the tuple object (array) but it's values (numbers).

**Comment by NoNameProvided** [comment]
I don't think we currently support tuples, but you can use [custom validation rules][1] to achieve the same goal.

[1]: https://github.com/typestack/class-validator#custom-validation-classes
